packages/plugin/src/VisualizationPlugin.js

```diff
@@ -113,34 +155,59 @@ export const VisualizationPlugin = ({
         return null
     }
 
-    if (
-        !fetchResult.visualization.type ||
-        fetchResult.visualization.type === VIS_TYPE_PIVOT_TABLE
-    ) {
-        return (
-            <PivotPlugin
-                visualization={fetchResult.visualization}
-                responses={fetchResult.responses}
-                legendSets={fetchResult.legendSets}
-                {...props}
-            />
-        )
-    } else {
-        return (
-            <ChartPlugin
-                visualization={fetchResult.visualization}
-                responses={fetchResult.responses}
-                extraOptions={fetchResult.extraOptions}
-                legendSets={fetchResult.legendSets}
-                {...props}
-            />
-        )
-    }
+    const contextualMenuRect =
+        contextualMenuRef &&
+        contextualMenuRef.current &&
+        contextualMenuRef.current.getBoundingClientRect()
```

Comment:
With `useRef`:
`const constextualMenuRect = contextualMenuRef.current?.getBoundingClientRect()`

Reply:
I tried this approach but it doesn't work for closing the contextual menu when clicking outside of it.
I think the reason is that the component does not re-render when the `contextualMenuRef.current` is reset and the check on `contextualMenuRect` done for showing the `ContextualMenu` component is not performed.
With `useState` instead the component re-renders when the state changes.

Reply:
Hmm... yeah I could see that being a problem.  I think it might be re-rendering any time the cell pivot table re-renders because otherwise I don't think the ref container would change?  But anyway not a big deal, this works as-is so fine for now, wasn't an important suggestion!

Reply:
You could change this to a single line with `?.` optional chaining, but totally optional (😆 )